interface/ceed-elemrestriction.c

```diff
@@ -185,6 +185,55 @@ int CeedElemRestrictionGetBackendStridesStatus(CeedElemRestriction rstr,
   return 0;
 }
 
+/**
+
+  @brief Get the E-vector layout of a CeedElemRestriction
+
+  @param rstr             CeedElemRestriction
+  @param[out] layout      Variable to store layout array,
+                            stored as [nodes, components, elements].
+                            The data for node i, component j, element k in the
+                            E-vector is given by
+                            i*strides[0] + j*strides[1] + k*strides[2]
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Backend
+**/
+int CeedElemRestrictionGetELayout(CeedElemRestriction rstr,
```

Comment:
At some point, it may make sense to have `CeedOperatorGetELayout` and `CeedOperatorGetQLayout` to better manage checking if the base and fallback operators are compatible.

Reply:
One of the problem I see here is that now the E-layout must be strided, if you look at the unused code for `cuda-reg`, the data layout expected was not strided.